src/wayland/output/handlers.rs

```diff
@@ -38,16 +38,22 @@ where
         data_init: &mut DataInit<'_, D>,
     ) {
         let client_scale = state.client_compositor_state(client).clone_client_scale();
-        let output = data_init.init(
+
+        let Some(output) = global_data.output.upgrade() else {
+            tracing::warn!("trying to bind a destroyed output, forgot to disable a output global?");
+            return;
```

Comment:
I don't think we should and can just return here. Any request to the `output` would otherwise trigger a panic, given we didn't initialize it's data.

The warning is fine (and a good idea), but I think we should just initialize the `OutputUserData` with `Weak::new`.